tests/unit/ng/event/test_event_state.py

```diff
@@ -141,3 +141,27 @@ def test_all_states_covered_in_state_machine() -> None:
     all_targets = {state for targets in graph.values() for state in targets}
     all_used = all_keys.union(all_targets)
     assert set(EventStateType).issubset(all_used)
+
+
+@pytest.mark.parametrize(
+    "state1, state2, expected",
+    [
+        (EventStateType.RECEIVING, EventStateType.RECEIVING, True),
+        (EventStateType.RECEIVING, EventStateType.RECEIVED, False),
+        (EventState(), EventState(), True),
+        (EventState(), EventStateType.RECEIVING, True),
+        (EventState(), EventStateType.DELIVERED, False),
+        (EventState(), object(), False),
+        (EventStateType.DELIVERED, "delivered", False),
+    ],
+)
+def test_equality(state1, state2, expected):
+    """Test equality of EventState instances."""
+    state_a = EventState() if isinstance(state1, EventStateType) else state1
+    state_b = EventState() if isinstance(state2, EventStateType) else state2
+    if isinstance(state1, EventStateType):
+        state_a.current_state = state1
+    if isinstance(state2, EventStateType):
+        state_b.current_state = state2
+    assert (state_a == state_b) is expected
+    assert (state_a != state_b) is not expected
```

Comment:
Maybe this could be me readable:

```suggestion
    """Test equality of EventState instances against various inputs."""

    def to_event_state(state):
        if isinstance(state, EventState):
            return state
        if isinstance(state, EventStateType):
            _state = EventState()  # <--- maybe as argument to __init__: EventState(starting_state=state) 
            _state.current_state = state
            return _state
        return state  # passt durch für object(), str etc.

    state_a = to_event_state(state1)
    state_b = to_event_state(state2)
    assert (state_a == state_b) is expected
    assert (state_a != state_b) is not expected
```

The to_event_state function could be moved to a utility module for reuse wherever state values need to be converted into EventState objects.